libpod/container.go

```diff
@@ -407,6 +409,58 @@ func (c *Container) teardownStorage() error {
 	return nil
 }
 
+func (c *Container) addImageData(g *generate.Generator) (err error) {
+	var (
+		args []string
+	)
+	if !c.config.UseImageConfig {
+		return nil
+	}
+	imageData, err := c.runtime.GetImageConfig(c.config.RootfsImageID)
+	if err != nil {
+		return errors.Wrapf(err, "error reading container image data")
+	}
+	if imageData.User != "" {
+		uid, _, err := chrootuser.LookupUserInContainer(c.bundlePath(), imageData.User)
+		if err != nil {
+			return errors.Wrapf(err, "user %q is not defined in %q", imageData.User, c.bundlePath())
+		}
+		g.SetProcessUID(uint32(uid))
+	}
+	for _, e := range imageData.ExposedPorts {
+		//TODO exposed ports are not handled yet
+		logrus.Warn("Exposed port %q is ignored", e)
+	}
+
+	if len(imageData.Cmd) > 0 {
+		args = imageData.Cmd
+	}
+	if len(imageData.Entrypoint) > 0 {
+		args = imageData.Entrypoint
+	}
+	if len(args) > 0 {
+		g.SetProcessArgs(args)
+	}
+	if imageData.StopSignal != "" {
+		g.AddAnnotation("org.opencontainers.image.stopSignal", imageData.StopSignal)
```

Comment:
Only do this if the user hasn't set it manually - if container.config.StopSignal isn't 0 we already have one set

Reply:
We should also do this when we create the container, not here. Probably near the end of runtime.NewContainer() before we dump it in the state, we should grab the image config, and if stopsignal is set and the user hasn't already set it through WithStopSignal, we set container.config.StopSignal

Reply:
This will remove the need for this code as the normal stop signal handling code will be used